src/AltinnCore/Common/Configuration/PlatformSettings.cs

```diff
@@ -10,16 +10,99 @@ namespace AltinnCore.Common.Configuration
         /// </summary>
-        public string GetApiBaseEndpoint()
+        public string GetApiStorageEndpoint
         {
-            return Environment.GetEnvironmentVariable("PlatformSettings__ApiBaseEndpoint") ?? ApiBaseEndpoint;
+            get
+            {
+                return Environment.GetEnvironmentVariable("PlatformSettings__ApiStorageEndpoint") ?? ApiStorageEndpoint;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the url host for API storage
+        /// </summary>
+        public string ApiStorageEndpointHost { get; set; }
+
+        /// <summary>
+        /// Gets the url host for API storage by looking into environment variables
+        /// </summary>
+        public string GetApiStorageEndpointHost
+        {
+            get
+            {
+                return Environment.GetEnvironmentVariable("PlatformSettings__ApiStorageEndpointHost") ?? ApiStorageEndpointHost;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the url for the API register endpoint
+        /// </summary>
+        public string ApiRegisterEndpoint { get; set; }
+
+        /// <summary>
+        /// Gets the url for the API register endpoint by looking into environment variables
+        /// </summary>
+        public string GetApiRegisterEndpoint
+        {
+            get
+            {
+                return Environment.GetEnvironmentVariable("PlatformSettings__ApiRegisterEndpoint") ?? ApiRegisterEndpoint;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the url host for API register
+        /// </summary>
+        public string ApiRegisterEndpointHost { get; set; }
+
+        /// <summary>
+        /// Gets the url host for API register by looking into environment variables
+        /// </summary>
+        public string GetApiRegisterEndpointHost
+        {
+            get
+            {
+                return Environment.GetEnvironmentVariable("PlatformSettings__ApiRegisterEndpointHost") ?? ApiRegisterEndpointHost;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the url for the API profile endpoint
+        /// </summary>
+        public string ApiProfileEndpoint { get; set; }
+
+        /// <summary>
+        /// Gets the url for the API profile endpoint by looking into environment variables
+        /// </summary>
+        public string GetApiProfileEndpoint
+        {
+            get
+            {
+                return Environment.GetEnvironmentVariable("PlatformSettings__ApiProfileEndpoint") ?? ApiProfileEndpoint;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the url host for API profile
+        /// </summary>
+        public string ApiProfileEndpointHost { get; set; }
+
+        /// <summary>
+        /// Gets the url host for API profile by looking into environment variables
+        /// </summary>
+        public string GetApiProfileEndpointHost
```

Comment:
do we need this method, cant we do this in the get of property f.ex

Reply:
This is needed to read from the kubernetes environment variable

Reply:
I meant why not move this             
get
            {
                return Environment.GetEnvironmentVariable("PlatformSettings__ApiProfileEndpointHost") ?? ApiProfileEndpointHost;
            }
to ApiProfileEndpointHost ?
So we can avoid having 1 more property

Reply:
you cannot reference ApiProfileEndpointHost in its own getter, I have tried the way you are describing and the code will not run if the code is that way